src/bigint-encoder.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Encode BigInt value from arbitrary provided arguments
+ *
+ * @param {Array<Number|BigInt|String>} parts - Slices to encode, where the
+ *    earlier elements represent higher bits in the final integer
+ * @param {64|128|256} size - Number of bits in the target integer type
+ * @param {Boolean} unsigned - Whether it's an unsigned integer
+ *
+ * @returns {BigInt}
+ */
+export function encodeBigIntFromBits(parts, size, unsigned) {
+  let result = 0n;
+  // check arguments length
+  if (parts.length && parts[0] instanceof Array) {
+    parts = parts[0];
+  }
+  const total = parts.length;
+  if (total === 1) {
+    try {
+      result = BigInt(parts[0]);
+      if (!unsigned) {
+        result = BigInt.asIntN(size, result);
+      }
+    } catch (e) {
+      throw new TypeError(`Invalid integer value: ${parts[0]}`)
+    }
+  } else {
+    const sliceSize = size / total;
+    if (sliceSize !== 32 && sliceSize !== 64 && sliceSize !== 128)
+      throw new TypeError('Invalid number of arguments')
+    // combine parts
+    for (let i = 0; i < total; i++) {
+      let part = BigInt.asUintN(sliceSize, BigInt(parts[i].valueOf()));
+      if (i > 0) { // shift if needed
+        part <<= BigInt(i * sliceSize);
+      }
+      result |= part;
+    }
+    if (!unsigned) { // clamp value to the requested size
+      result = BigInt.asIntN(size, result);
+    }
+  }
+  // check type
+  if (typeof result === 'bigint') {
+    // check boundaries
+    const [min, max] = calculateBigIntBoundaries(size, unsigned);
+    if (result >= min && result <= max)
+      return result;
+  }
+  // failed to encode
+  throw new TypeError(`Invalid ${formatIntName(size, unsigned)} value`);
+}
+
+/**
+ * @param {BigInt} value - Slices to encode
+ * @param {64|128|256} size - Number of bits in the source integer type
+ * @param {32|64|128} sliceSize - Number of parts
+ * @return {BigInt[]}
+ */
+export function sliceBigInt(value, size, sliceSize) {
+  if (typeof value !== 'bigint')
+    throw new TypeError('Invalid BigInt value');
+  const total = size / sliceSize;
+  if (total === 1)
+    return [value];
+  if (sliceSize < 32 || sliceSize > 128 || total !== 2 && total !== 4 && total !== 8)
```

Comment:
```suggestion
  if (sliceSize < 32 || sliceSize > 128 || (total !== 2 && total !== 4 && total !== 8))
```

Reply:
That change doesn't alter the internal logic. Is this a code-style rule?